src/undo/LineStyleUndoAction.cpp

```diff
@@ -0,0 +1,86 @@
+#include "LineStyleUndoAction.h"
+
+#include "gui/Redrawable.h"
+#include "model/PageRef.h"
+#include "model/Stroke.h"
+
+#include "Rectangle.h"
+#include "i18n.h"
+
+class LineStyleUndoActionEntry {
+public:
+    LineStyleUndoActionEntry(Stroke* s, LineStyle oldStyle, LineStyle newStyle):
+            s(s), oldStyle(oldStyle), newStyle(newStyle) {}
+    Stroke* s;
+    LineStyle oldStyle;
+    LineStyle newStyle;
+};
+
+LineStyleUndoAction::LineStyleUndoAction(const PageRef& page, Layer* layer): UndoAction("LineStyleUndoAction") {
+    this->page = page;
+    this->layer = layer;
+}
+
+LineStyleUndoAction::~LineStyleUndoAction() {
+    for (LineStyleUndoActionEntry* e: this->data) {
+        delete e;
+    }
+}
```

Comment:
```suggestion
```
Since you forwarded LineStyleUndoActionEntry, it could be required to default the destructor here.:
```suggestion
LineStyleUndoAction::~LineStyleUndoAction() =default;
```